pkg/reconciler/v1alpha1/taskrun/resources/pod.go

```diff
@@ -248,20 +285,56 @@ func MakePod(taskRun *v1alpha1.TaskRun, taskSpec v1alpha1.TaskSpec, kubeclient k
 			Annotations: annotations,
 			Labels:      makeLabels(taskRun),
 		},
-		Spec: corev1.PodSpec{
-			// If the build fails, don't restart it.
-			RestartPolicy:      corev1.RestartPolicyNever,
-			InitContainers:     initContainers,
-			Containers:         podContainers,
-			ServiceAccountName: taskRun.Spec.ServiceAccount,
-			Volumes:            volumes,
-			NodeSelector:       taskRun.Spec.NodeSelector,
-			Tolerations:        taskRun.Spec.Tolerations,
-			Affinity:           taskRun.Spec.Affinity,
-		},
+		Spec: podSpec,
 	}, nil
 }
 
+
+func mergeStepWithBaseContainer(base, override *corev1.Container) (*corev1.Container, error) {
```

Comment:
I think we could have this in it's own package with unit tests :D

(No worries if you want to wait to till we decide if we want to go forward with this impl vs the previous one)

Reply:
=)

Reply:
While I'm thinking of it, a question - what's the motivation for putting functions like this in their own packages, rather than just unit testing them in place? Just trying to understand. =)